src/main/java/in/projecteka/consentmanager/user/LockedUserService.java

```diff
@@ -25,12 +25,16 @@
         return lockedUsersRepository.deleteUser(cmId);
     }
 
-    public Mono<Void> createOrUpdateLockedUser(String cmId) {
+    public Mono<Integer> createOrUpdateLockedUser(String cmId) {
         return lockedUsersRepository.getLockedUserFor(cmId)
-                .flatMap(lockedUser ->
-                        isBeforeMinutes(lockedUser.getDateCreated(), lockedServiceProperties.getCoolOfPeriod())
-                                ? Mono.empty() : removeLockedUser(cmId))
-                .then(lockedUsersRepository.upsert(cmId));
+                .flatMap(lockedUser -> {
+                    var remainingTries = lockedServiceProperties.getMaximumInvalidAttempts() - lockedUser.getInvalidAttempts();
+                    return isBeforeMinutes(lockedUser.getDateCreated(), lockedServiceProperties.getCoolOfPeriod())
+                            ? Mono.just(remainingTries - 1)
+                            : removeLockedUser(cmId).thenReturn(lockedServiceProperties.getMaximumInvalidAttempts() - 1);
```

Comment:
whenever you have ternary operator, that's a sign for possible filter operator.

Reply:
not sure how can I do that in this case. If you have any suggestion, please tell me

Reply:
in consent manager.cs we are doing it in many places. Refer those please. If you can't, ignore this.

Reply:
.cs in c#? which service?

Reply:
sorry ConsentManager.java